### PR TITLE
feat: per-set duration in seconds, exercise types, and cardio distance

### LINKED ISSUE #1903: [Bug]: Workout Preset Duration Units Different in Android App
### What happened?

Hello - just getting going with the app. So first, thank you for this amazing tool!

When creating a workout preset in the webui, the duration is requested in minutes. When I go to that preset in the Android app, it is reported in seconds (ie: I put 1.5 minutes in and the app shows as a 1.5 second duration). 

### Steps to reproduce

1. Create workout preset in webUI
2. Set duration to something like 1.5 - note the claimed unit is minutes
3. Save workout preset
4. Go to workout preset in Android app
5. Notice that the duration is reported in seconds

### Expected behavior

The duration to be reported in a uniform unit

### How are you running SparkyFitness?

Other

### SparkyFitness version

v0.17.3

### Which area is affected?

Mobile app (Android)

### Docker Compose & Environment Configuration

```yaml

```

### Runtime Logs (Server, Web, or Mobile)

```shell

```

### Screenshots or Screen Recordings

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by apedley**
> [!TIP]
> **Help us review and merge your PR faster!**
> Please ensure you have completed the **Checklist** below.
> For **Frontend** changes, please run `pnpm run validate` to check for any errors.
> PRs that include tests and clear screenshots are highly preferred!
> Note: AI-generated descriptions must be manually edited for conciseness. Do not paste raw AI summaries.

## Description

**What problem does this PR solve?**
Duration units didn't match between web and mobile for exercise tracking. Web used minutes.

**How did you implement the solution?**
(Brief technical approach.)

Per-set duration finally gets a real unit, exercises get an explicit tracking type, and cardio gets logged as a set carrying duration + distance.

1. Per set duration is now integer seconds (was unitless numeric and web wrote fractional minutes while chatbot and mobile logged seconds). All integrations, labels, outputs, and everything else (I really hope) has been updated for this.
2. Added modality to exercises. Currently weight_reps | reps_only | duration | duration_distance. This selects which set editor is displayed in the UI so planks/holds get a proper inputs for seconds instead of hiding it in reps.
3. Cardio is set backed - a  duration_distance entry stores its effort on a single set (duration in seconds and a new distance in km column) while the UI still presets as a simple duration and distance form.

One migration was added and it's a doozy. I had to migrate old data that was stored in minutes to seconds. Because some data might have been entered in seconds, I've created a new *system* table (doesn't use rls) system.set_duration_premigration_backup to store the old data. I really don't expect any problems with this but I'm being extra careful since I hate migrating user data like this. However the migration is atomic and idempotent so repeat runs won't cause issues and the backup table is a safety net for repair in case there are reported issues. I asked Claude how risky this was and it said a 4/10.

Linked Issue: Closes #1903

## How to Test

1. Check out this branch and run `...`
2. Navigate to...
4. Verify that...

## PR Type

- [x] Issue (bug fix)
- [x] New Feature
- [x] Refactor
- [ ] Documentation

## Checklist

**All PRs:**

- [x] **[MANDATORY - ALL] Integrity & License**: I certify this is my own work, free of malicious code, and I agree to the [License terms](https://github.com/CodeWithCJ/SparkyFitness/blob/main/LICENSE).

**New features only:**

- [x] **[MANDATORY for new feature] Alignment**: I have raised a GitHub issue and it was reviewed/approved by maintainers or it was approved on Discord.

**Frontend changes (`SparkyFitnessFrontend/`):**

- [x] **[MANDATORY for Frontend changes] Quality**: I have run `pnpm run validate` and it passes.
- [x] **[MANDATORY for Frontend changes] Translations**: I have only updated the English (`en`) translation file.

**Backend changes (`SparkyFitnessServer/`):**

- [x] **[MANDATORY for Backend changes] Code Quality**: I have run typecheck, lint, and tests. New files use TypeScript, new endpoints have Zod schemas, and new endpoints include tests.
- [x] **[MANDATORY for Backend changes] Database Security**: I have updated `rls_policies.sql` for any new user-specific tables.

**UI changes (components, screens, pages):**

- [ ] **[MANDATORY for UI changes] Screenshots**: I have attached Before/After screenshots below.

**Mobile changes (`SparkyFitnessMobile/`):**

- [x] **[MANDATORY for Mobile changes] Tested on device or emulator**: I have verified the changes work on iOS or Android.

## Screenshots

3 different types of set:

image

<details>
<summary>Click to expand</summary>

### Before

before

### After

after

</details>

## Notes for Reviewers

> Optional — use this for anything that doesn't fit above: known tradeoffs, areas you'd like specific feedback on, qustions you have or context that helps reviewers.

## Summary by CodeRabbit

* **New Features**
  * Added exercise “tracking types” (weight/reps, reps-only, duration, duration+distance) with modality-aware set tables and entry/history displays.
  * Added per-set duration (seconds) and per-set distance support with km/miles unit conversion for cardio.
* **Bug Fixes**
  * Preserved true “no weight” as empty (null) and fixed distance clearing when the value is removed.
  * Improved cardio/legacy set handling, duration aggregation, and reporting output (e.g., avoiding “NaN”/“0 - 0”).
* **Documentation**
  * Updated health-data sync documentation for per-set duration units and legacy behavior.
* **Tests**
  * Added/expanded unit and UI coverage for modality, cardio effort, duration/distance, and edge cases.

### CODE REVIEW

**Comment by apedley** [comment]
@coderabbitai review

File: SparkyFitnessServer/db/migrations/20260727100000_set_duration_seconds_modality_distance.sql [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    IF EXISTS (
+        SELECT 1 FROM information_schema.columns
+        WHERE table_schema = 'public' AND table_name = 'workout_preset_exercise_sets'
+        AND column_name = 'duration' AND data_type = 'numeric'
+    ) THEN
+        ALTER TABLE workout_preset_exercise_sets
+        ALTER COLUMN duration TYPE integer USING ROUND(duration * 60)::integer;
+    END IF;
+END $$;
```
[the remark this answers is not in the document]
Comment (apedley) [reply SparkyFitnessServer/db/migrations/20260727100000_set_duration_seconds_modality_distance.sql:56]: @coderabbitai scenario needs a DB where exercise_entry_sets.duration is already integer while the other two tables are still numeric. That state can't exist. All of them are set together in a DO $$ block.

File: SparkyFitnessMobile/__tests__/hooks/useWorkoutForm.test.ts [hunk 2]
```diff
@@ -729,6 +793,39 @@ describe('workoutFormReducer', () => {
...
+      // Old-server sessions omit it — the draft records null and resolvers
+      // fall back to the category.
+      const bare = workoutFormReducer(state, {
+        type: 'POPULATE',
+        session: makeSession(),
+        weightUnit: 'kg',
+      });
+      expect(bare.exercises[0].exerciseModality).toBeNull();
+    });
+
```
[the remark this answers is not in the document]
Comment (apedley) [reply SparkyFitnessMobile/__tests__/hooks/useWorkoutForm.test.ts:829]: pnpm run typecheck exits 0 because tsconfig.json excludes __tests__ meaning this won't break CI.

**Review by CodeWithCJ — APPROVED** [review]
